### DISCUSSION TITLE
quarto vscode extension enforcing a higher than expected quarto version

### DISCUSSION THREAD

**Discussion by boshek** [quarto-dev/quarto-cli#10143] [Q&A] [answered]
### Description

Does anyone have any thoughts why my quarto vscode extension is suddenly requiring a specific version of quarto? 

This is what I have with my default keybinding to render:
```
{ "key": "shift+cmd+k",           "command": "quarto.preview",
                                     "when": "!quartoRenderScriptActive" },
```

when I hit `shift+cmd+k` I get this popup:

![image](https://github.com/quarto-dev/quarto-cli/assets/18472598/a04188b7-3eb6-4d7e-a589-9344d5c9710d)

which is executing this code path:

https://github.com/quarto-dev/quarto/blob/06678d55143a7d6de6c7f0232db0c5dcf8392ca6/apps/vscode/src/providers/render.ts#L57-L66

which suggest that version `0.9.149` is the minimum version. Bumping to version 1.5.39 does indeed fix this but I'm not able to yet go to that preview version. 

quarto itself still works great from the CLI with `quarto preview`, `quarto render`, etc all still working fine. It is just that my keyboard shortcut is not working. AFAIK, I did not modify any quarto setting. As an additional data point, if I am working on a qmd with an active preview (invoke by `quarto preview file.qmd`) _then_ I hit the keyboard shortcut it will re-render the active file. 🤔 

Things I've tried:
- re-install vscode extension. 
- install newer version of quarto (works but not a workable solution at the moment)
- invoke same file from RStudio (works)

Has this ever happened to anyone before? 

### Version Info
quarto version: `1.4.555`
quarto vscode extension version: `1.113.0`
vscode version `1.90`
macOS: 14.5

**Comment by mcanouil** [comment]
Install a previous version of the extension.
<img width="677" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/b4309239-0e45-4334-adb9-ea542fdfa17f">

**Reply by mcanouil** [reply, marked as answer]
FYI, this was introduced in https://github.com/quarto-dev/quarto/commit/06678d55143a7d6de6c7f0232db0c5dcf8392ca6 

The rational is for R package support: https://github.com/quarto-dev/quarto/pull/455

@boshek Are you in an R package workspace? (A workspace with a `DESCRIPTION` file with `Package` https://github.com/quarto-dev/quarto/blob/06678d55143a7d6de6c7f0232db0c5dcf8392ca6/apps/vscode/src/providers/preview/preview-util.ts#L62-L75)

**Reply by boshek** [reply]
Thank you for the response. Much appreciated. 

Yes I am in a package workspace but used as a project. So the folder structure does have a `DESCRIPTION` file and tests and functions in a `R` but the package is never intended to be installed. It is just leveraging the package features for organization. 

From https://github.com/quarto-dev/quarto/pull/435 I am seeing that this use case hasn't been considered. Looks like I can turn off the feature so that's ok, though I agree very much with your comment [here](https://github.com/quarto-dev/quarto/commit/06678d55143a7d6de6c7f0232db0c5dcf8392ca6#r143578146). Not sure why we are defaulting to a pre-release package like this.

**Reply by cscheid** [reply]
> Not sure why we are defaulting to a pre-release package like this.

I hear you. There is a reason for it that we're not entirely ready to talk about in public, so please bear that in mind for a few weeks.

**Reply by boshek** [reply]
Gotcha. And that wasn't meant to be snark. Thanks for everything!

**Reply by cscheid** [reply]
Ok, I can now explain, a bit earlier than expected.

The reason for that version setting is that it's required for Positron: https://github.com/posit-dev/positron/

**Reply by boshek** [reply]
OMG. I feel like this was built specifically for me. What a joy.

**Reply by boshek** [reply]
@cscheid is there any way to disable this check? I use a `DESCRIPTION` file outside the package context and so don't want quarto to behave as though it is an R package.

**Reply by mcanouil** [reply]
@boshek As shown, you have to downgrade the extension. There are no other choices right now unfortunately.
https://github.com/quarto-dev/quarto-cli/discussions/10143#discussioncomment-9887306

See the changelog: https://marketplace.visualstudio.com/items/quarto.quarto/changelog

Edit: what is inside your `DESCRIPTION` file?
Do you have `Package: XXXXX`? This would mean "it's an R package".

**Reply by boshek** [reply]
Ok I just wanted to report one additional thing here for anyone that might come across this discussion. You _can_ actually disable rendering to a temporary directory if you want as [mentioned here](https://github.com/quarto-dev/quarto/pull/435#pullrequestreview-2046850083).
